### ISSUE TITLE
[Bug]: memory access out of bounds

### ISSUE THREAD

**Issue by JoanDuatis** [IFCjs/web-ifc#448]
### What happened?

Hello,

I'm trying to load a 845,000 kb model in an application that doesn't need to display the geometry.
Therefore I am using only web-ifc.

During the loading process a "memory access out of bounds" error occurs.

However, in another application that I have made with web-ifc-viewer, it loads it without problems.

I am using version 0.0.41 of web-ifc but I have also tried 0.0.40 and 0.0.39 without success.

The javascript code that I am using to open the file is the following:!

async function loadIfc() {
    const file = input.files[0];
    const url = URL.createObjectURL(file);
    fetch(url)
    .then(response => response.text())
    .then(data => {
        LoadFileData(data);
    });
}
async function LoadFileData(ifcAsText) {
    const uint8array = new TextEncoder().encode(ifcAsText);
    const modelID = await OpenIfc(uint8array);
    modelsIDs.push(modelID);
    GetAllClassesOfProducts(modelID);
   filter(modelID);
}

async function OpenIfc(ifcAsText) {
    await ifcapi.Init();
    return ifcapi.OpenModel(ifcAsText,   { OPTIMIZE_PROFILES: true, 
        COORDINATE_TO_ORIGIN: true, 
        USE_FAST_BOOLS: true });
}

The javascript code that I am using to open the file is the following:

This is the error that appears:
![image](https://github.com/IFCjs/web-ifc/assets/130443866/ef7a2b99-efa5-4952-9eac-32c8b2e79e71)

### Version

0.0.41

### What browsers are you seeing the problem on?

Firefox, Chrome

**Comment by JoanDuatis** [comment]
It can be reproduced with the same file as the issue https://github.com/IFCjs/web-ifc/issues/374

**Comment by beachtom** [comment]
So I think this is already fixed. Can you try using the current download from the readme file and extracting into node_modules/web-ifc

I tried it with that and it worked fine.

**Comment by JoanDuatis** [comment]
I have followed the indicated steps and I have not been able to load large files using only web-ifc. However with WIV it works fine.

I have unzipped in node_modules/web-ifc the latest version that is downloaded from the Readme file. I have made a new build and it doesn't work. I have copied the wasm files from node_modules/web-ifc to the kroot of my project, new build and new test that doesn't work.

So I have gone to replicate the load procedure that is exposed in the readme file but the use of require of the first line gives me an error:

const WebIFC = require("web-ifc/web-ifc-api.js");
const ifcApi = new WebIFC.IfcAPI();

so i put it back like this:

import { IfcAPI} from "web-ifc";
const ifcapi = new IfcAPI();

I admit my ignorance (I am a beginner) and perhaps what I have done is not equivalent and that is why it does not work. But then why does it work with small or medium files?

I have followed the indicated steps and I have not been able to load large files using only web-ifc. However with WIV it works fine.

I have unzipped in node_modules/web-ifc the latest version that is downloaded from the Readme file. I have made a new build and it doesn't work. I have copied the wasm files from node_modules/web-ifc to the kroot of my project, new build and new test that doesn't work.

So I have gone to replicate the load procedure that is exposed in the readme file but the use of require of the first line gives me an error:

const WebIFC = require("web-ifc/web-ifc-api.js");
const ifcApi = new WebIFC.IfcAPI();

so i put it back like this:

import { IfcAPI} from "web-ifc";
const ifcapi = new IfcAPI();

I admit my ignorance (I am a beginner) and perhaps what I have done is not equivalent and that is why it does not work. But then why does it work with small or medium files?

[VinculacioIFCPOs.zip](https://github.com/IFCjs/web-ifc/files/12084136/VinculacioIFCPOs.zip)
